src/app/model-download/components/model-download-component/model-download-component.component.ts

```diff
@@ -0,0 +1,172 @@
+import { Component, OnInit } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { ActivatedRoute } from '@angular/router';
+import { Subscription, timer } from 'rxjs';
+import { first } from 'rxjs/operators';
+import { ConfigManager } from 'src/app/base/services/config-service';
+import { NotificationService } from 'src/app/base/services/notification.service';
+import { ProjectApolloService } from 'src/app/base/services/project/project-apollo.service';
+import { RouteService } from 'src/app/base/services/route.service';
+import { WeakSourceApolloService } from 'src/app/base/services/weak-source/weak-source-apollo.service';
+import { dateAsUTCDate } from 'src/app/util/helper-functions';
+
+@Component({
+  selector: 'kern-model-download-component',
+  templateUrl: './model-download-component.component.html',
+  styleUrls: ['./model-download-component.component.scss']
+})
+export class ModelDownloadComponentComponent implements OnInit {
+
+  project$: any;
+  projectQuery$: any;
+  downloadedModelsList$: any;
+  downloadedModelsQuery$: any;
+  projectId: string;
+  form: FormGroup;
+  models: any[] = [];
+  isManaged: boolean = false;
+  currentModelHandle: any;
+  downloadedModels: any[] = [];
+  subscriptions$: Subscription[] = [];
+
+  constructor(
+    private activatedRoute: ActivatedRoute,
+    private projectApolloService: ProjectApolloService,
+    private routeService: RouteService,
+    private informationSourceApolloService: WeakSourceApolloService,
+    private formBuilder: FormBuilder
+  ) { }
+
+  ngOnInit(): void {
+    this.initForm();
+    this.routeService.updateActivatedRoute(this.activatedRoute);
+    this.projectId = this.activatedRoute.parent.snapshot.paramMap.get('projectId');
+    [this.projectQuery$, this.project$] = this.projectApolloService.getProjectByIdQuery(this.projectId);
+    [this.downloadedModelsQuery$, this.downloadedModelsList$] = this.informationSourceApolloService.getModelProviderInfo();
+    this.prepareModels();
+    this.prepareZeroShotRecommendations(this.projectId);
+    this.isManaged = ConfigManager.getIsManaged();
+
+    this.subscriptions$.push(
+      this.downloadedModelsList$.subscribe((downloadedModels) => this.downloadedModels = downloadedModels));
+
+    NotificationService.subscribeToNotification(this, {
+      projectId: this.projectId,
+      whitelist: ['model_provider_download'],
+      func: this.handleWebsocketNotification
+    });
+  }
+
+  ngOnDestroy(): void {
+    this.subscriptions$.forEach((subscription) => subscription.unsubscribe());
+    NotificationService.unsubscribeFromNotification(this, this.projectId)
+  }
+
+  initForm() {
+    this.form = this.formBuilder.group({
+      name: ['', Validators.required]
+    })
+  }
+
+  deleteModel(name: string, revision: string) {
+    this.subscriptions$.push(
```

Comment:
no need to push this to subscriptions since the pipe(first()) unsubscribes on it's own